source/extensions/filters/http/response_map/response_map_filter.cc

```diff
@@ -26,14 +29,35 @@ ResponseMapFilter::ResponseMapFilter(ResponseMapFilterConfigSharedPtr config)
 Http::FilterHeadersStatus ResponseMapFilter::decodeHeaders(Http::RequestHeaderMap& request_headers,
                                                            bool end_stream) {
   ENVOY_LOG(trace, "response map filter: decodeHeaders with end_stream = {}", end_stream);
+
+  // Disable filter per route config if applies
+  if (decoder_callbacks_->route() != nullptr) {
+    const auto* per_route_config =
+        Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(
+            Extensions::HttpFilters::HttpFilterNames::get().ResponseMap,
+            decoder_callbacks_->route());
+    ENVOY_LOG(trace, "response map filter: found route. has per_route_config? {}",
+        per_route_config != nullptr);
+    if (per_route_config != nullptr && per_route_config->disabled()) {
+      ENVOY_LOG(trace, "response map filter: disabling due to per_route_config");
+      disabled_ = true;
+      return Http::FilterHeadersStatus::Continue;
+    }
+  }
+
   request_headers_ = &request_headers;
   return Http::FilterHeadersStatus::Continue;
 }
 
 Http::FilterHeadersStatus ResponseMapFilter::encodeHeaders(Http::ResponseHeaderMap& headers,
                                                            bool end_stream) {
-  ENVOY_LOG(trace, "response map filter: encodeHeaders with http status = {} and end_stream = {}",
-      headers.getStatusValue(), end_stream);
+  ENVOY_LOG(trace, "response map filter: encodeHeaders with http status = {}, end_stream = {}, disabled = {}",
```

Comment:
Should that be `, and disabled =`? IDK what style is, I'm just noting that there was an `and` there before.

Reply:
i removed the `and` in favor of comma separating the 3 values